src/test/java/htsjdk/samtools/CRAMComplianceTest.java

```diff
@@ -261,6 +261,39 @@ public void testBAMThroughCRAMRoundTrip() throws IOException, NoSuchAlgorithmExc
         }
     }
 
+    @Test
+    public void testBAMThroughCRAMRoundTripViaPath() throws IOException, NoSuchAlgorithmException {
+        final File TEST_DATA_DIR = new File("src/test/resources/htsjdk/samtools/cram");
+
+        // These files are reduced versions of the CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam and human_g1k_v37.20.21.fasta
+        // files used in GATK4 tests. The first 8000 records from chr20 were extracted; from those around 80 placed but
+        // unmapped reads that contained cigar elements were removed, along with one read who's mate was on chr21.
+        // Finally all read positions were remapped to the subsetted reference file, which contains only the ~9000 bases
+        // used by the reduced read set.
+        final File originalBAMInputFile = new File(TEST_DATA_DIR, "CEUTrio.HiSeq.WGS.b37.NA12878.20.first.8000.bam");
+        final File referenceFile = new File(TEST_DATA_DIR, "human_g1k_v37.20.subset.fasta");
+
+        // retrieve all records from the bam and reset the indexing bins to keep comparisons with
+        // cram records from failing
+        List<SAMRecord> originalBAMRecords = getSAMRecordsFromFile(originalBAMInputFile, referenceFile);
+        for (int i = 0; i < originalBAMRecords.size(); i++) {
+            originalBAMRecords.get(i).setIndexingBin(null);
+        }
+
+        // write the BAM records to a temporary CRAM
+        final File tempCRAMFile = File.createTempFile("testBAMThroughCRAMRoundTrip", CramIO.CRAM_FILE_EXTENSION);
```

Comment:
My suggestion was related to use at least in the changed test the jims filesystem, but anyway I think that some of the tests should still have `File`, which is still quite common.